### PR TITLE
Node policy integration

### PR DESCRIPTION
**Pull request by danielsdeleo**
Integrate node objects and policyfile attributes by updating node create/update to accept `policy_group` and `policy_name`

These things are dependent but will be done later to keep patch size small:
* add endpoint `GET /policies/:policy_name/nodes`
* add endpoint `GET /policy_groups/:policy_group/nodes`

### CODE REVIEW

**Comment by danielsdeleo** [comment]
@chef/lob not complete yet, but could use review on the schema stuff especially.

**Comment by danielsdeleo** [comment]
BTW, Adding the two columns to the nodes table didn't seem to make much difference in testing against preprod: https://gist.github.com/danielsdeleo/53154586743b4dbbe217

File: src/oc_erchef/apps/chef_objects/src/chef_node.erl [hunk 1]
```diff
@@ -76,12 +77,16 @@
...
          ]}).
 
 -define(VALID_KEYS,
         [<<"name">>, <<"chef_environment">>, <<"json_class">>, <<"chef_type">>,
-         <<"normal">>, <<"default">>, <<"override">>, <<"automatic">>, <<"run_list">>]).
+         <<"normal">>, <<"default">>, <<"override">>, <<"automatic">>, <<"run_list">>,
+         <<"policy_name">>, <<"policy_group">> ]).
 
 -define(create_if_missing_fields,
         [{<<"chef_environment">>, <<"_default">>},
```
Comment (marcparadise) [inline src/oc_erchef/apps/chef_objects/src/chef_node.erl]: Do we want to provide defaults for the new fields?

Comment (danielsdeleo) [inline src/oc_erchef/apps/chef_objects/src/chef_node.erl]: I don't think there is a sane default. I was planning on having `null` for these fields be an indication to `chef-client` that it's running in "not policyfiles mode". The only thing I think I could put as a default is something that would be invalid for the user to upload, like `!nothing` but that leads to a lot of ugly elsewhere, I think.

**Comment by tylercloke** [comment]
Looks good!

**Comment by marcparadise** [comment]
:+1:

**Pull request merged**